src/Functions/FunctionFactory.h

```diff
@@ -82,3 +82,14 @@ class FunctionFactory : private boost::noncopyable,
 };
 
 }
+
+typedef void (*func_entry)(DB::FunctionFactory &factory);
+
+/// --- Internal macros, don't use
+/// --- only functions name start with 'register' will be called
+#define FUNCTION_REGISTER_IMPL(fn) \
+__attribute__((cold, used)) void fn(FunctionFactory & factory)
+
+/// --- Macros for registering functions
+/// --- always add 'register' prefix
+#define FUNCTION_REGISTER(fn) FUNCTION_REGISTER_IMPL(register##fn)
```

Comment:
Perhaps add an "underscore" in front of "register?" to indicate it's truly an internal function?